app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/EnabledMessageComposer.kt

```diff
@@ -0,0 +1,266 @@
+import androidx.compose.animation.animateContentSize
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ExperimentalLayoutApi
+import androidx.compose.foundation.layout.WindowInsets
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.ime
+import androidx.compose.foundation.layout.isImeVisible
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.unit.dp
+import com.wire.android.ui.common.banner.SecurityClassificationBanner
+import com.wire.android.ui.common.bottombar.BottomNavigationBarHeight
+import com.wire.android.ui.common.colorsScheme
+import com.wire.android.ui.home.conversations.model.UriAsset
+import com.wire.android.ui.home.messagecomposer.state.AdditionalOptionSelectItem
+import com.wire.android.ui.home.messagecomposer.state.AdditionalOptionSubMenuState
+import com.wire.android.ui.home.messagecomposer.state.MessageComposerStateHolder
+import com.wire.android.ui.home.messagecomposer.state.MessageCompositionType
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.util.isPositiveNotNull
+
+@OptIn(ExperimentalLayoutApi::class)
+@Suppress("ComplexMethod")
+@Composable
+fun EnabledMessageComposer(
+    conversationId: ConversationId,
+    messageComposerStateHolder: MessageComposerStateHolder,
+    snackbarHostState: SnackbarHostState,
+    messageListContent: @Composable () -> Unit,
+    onChangeSelfDeletionClicked: () -> Unit,
+    onSearchMentionQueryChanged: (String) -> Unit,
+    onSendButtonClicked: () -> Unit,
+    onAttachmentPicked: (UriAsset) -> Unit,
+    onAudioRecorded: (UriAsset) -> Unit,
+    onPingOptionClicked: () -> Unit,
+    onClearMentionSearchResult: () -> Unit,
+    tempWritableVideoUri: Uri?,
+    tempWritableImageUri: Uri?
+) {
+    val density = LocalDensity.current
+    val focusManager = LocalFocusManager.current
+    val navBarHeight = BottomNavigationBarHeight()
+    val isImeVisible = WindowInsets.isImeVisible
+    val offsetY = WindowInsets.ime.getBottom(density)
+
+    with(messageComposerStateHolder) {
+        val inputStateHolder = messageCompositionInputStateHolder
+
+        LaunchedEffect(offsetY) {
```

Comment:
question: doesn't it happen too often? I think on newer Android versions `WindoeInsets.ime` is quite dynamic to allow for the animations so it can be updated with each frame/recomposition - wouldn't be better to move calculations like this and handling offset changes to the layout phase to prevent from too many recompositions?